CRM/CiviDiscount/DiscountCalculator.php

```diff
@@ -182,6 +182,12 @@ function setEntityDiscounts() {
       }
       if ($this->checkDiscountsByEntity($discount, $this->entity, $this->entity_id, 'filters')) {
         $this->entity_discounts[$discount_id] = $discount;
+        /* Added by BOT
+        * To decide if discount text box should be displayed, no need to loop through all discount codes 
+        */ 
+        if( empty( $this->code ) ) {
+          break;
+        }
```

Comment:
please check [our code style guide](http://wiki.civicrm.org/confluence/display/CRMDOC/PHP+Code+and+Inline+Documentation)

* inline comments should start with two slashes not `/*`
* there should be a space after the word `if`
* there should be no space around opening and closing parentheses